dataprofiler/tests/space_time_analysis/structured_space_time_analysis.py

```diff
@@ -0,0 +1,234 @@
+except ImportError:
+    import dataprofiler as dp
+
+from dataset_generation import generate_dataset_by_class
+
+from dataprofiler import StructuredProfiler
+
+# suppress TF warnings
+tf.compat.v1.logging.set_verbosity(tf.compat.v1.logging.ERROR)
+
+
+def dp_profile_space_analysis(
+    data: pd.DataFrame, path: str, options: Dict = None, percent_to_nan: float = 0.0
+) -> StructuredProfiler:
+    """
+    Generate memray bin file of the space analysis of dp.Profiler function
+
+    :param data: DataFrame that is to be profiled
+    :type data: pandas.DataFrame
+    :param path: Path to output the memray bin file generated for space analysis
+    :type path: string
+    :param options: options for the dataprofiler intialization
+    :type options: Dict, None, optional
+    :param percent_to_nan: Percentage of dataset that needs to be nan values
+    :type percent_to_nan: float, optional
+
+    :return: The StructuredProfile generated by dp.Profiler
+    """
+    if percent_to_nan:
+        nan_injection(df, percent_to_nan)
+    with memray.Tracker(path):
+        profile = dp.Profiler(data, options=options, samples_per_update=len(data))
+    return profile
+
+
+def dp_merge_space_analysis(profile: StructuredProfiler, path: str):
+    """
+    Generate memray bin file of the space analysis of merge profile functionality
+
+    :param profile: Profile that is to be merged with itself
+    :type profile: StructuredProfile
+    :param path: Path to output the memray bin file generated for space analysis
+    :type path: string
+    """
+
+    with memray.Tracker(path):
+        _ = profile + profile
+
+
+def dp_time_analysis(
+    sample_sizes: List,
+    data: pd.DataFrame,
+    path: str = "structured_profiler_times.json",
+    percent_to_nan: float = 0.0,
+    allow_subsampling: bool = True,
+    options: Dict = None,
+):
+    """
+    Run time analysis for profile and merge functionality
+
+    :param sample_sizes: List of sample sizes of dataset to be analyzed
+    :type sample_sizes: list
+    :param data: DataFrame to be used for time analysis
+    :type data: pandas DataFrame
+    :param path: Path to output json file with all time analysis info
+    :type path: string, optional
+    :param percent_to_nan: Percentage of dataset that needs to be nan values
+    :type percent_to_nan: float, optional
+    :param allow_subsampling: boolean to allow subsampling when running time analysis
+    :type allow_subsampling: bool, optional
+    :param options: options for the dataprofiler intialization
+    :type options: Dict, None, optional
+
+    """
+    # [0] allows model to be initialized and added to labeler
+    sample_sizes = [0] + sample_sizes
+    profile_times = []
+    for sample_size in sample_sizes:
+        # setup time dict
+
+        print(f"Evaluating sample size: {sample_size}")
+        df = data.sample(sample_size, replace=True).reset_index(drop=True)
+
+        if percent_to_nan:
+            nan_injection(df)
```

Comment:
`df = nan_injection(df)`

i know it is mutable and is working, but might be safer.